src/libraries/System.Security.Cryptography.X509Certificates/src/Microsoft/Win32/SafeHandles/SafePasswordHandle.cs

```diff
@@ -25,6 +25,26 @@ public SafePasswordHandle(string? password)
             }
         }
 
+        public SafePasswordHandle(ReadOnlySpan<char> password)
+            : base(ownsHandle: true)
+        {
+            // "".AsSpan() is not default, so this is compat for "null tries NULL first".
+            if (password != default)
+            {
+                int spanLen = password.Length + 1;
```

Comment:
`(span.Length + 1) * sizeof(char)` could in theory integer overflow. Though why anybody would have a 2bn-element password I couldn't say. :)